pkgs/tools/package-management/libdnf/default.nix

```diff
@@ -0,0 +1,62 @@
+{ gcc9Stdenv, stdenv, fetchFromGitHub, cmake, gettext, pkgconfig, gpgme, libsolv, openssl, check, pcre, json_c, libmodulemd, libsmartcols, sqlite, librepo, libyaml, rpm }:
+
+gcc9Stdenv.mkDerivation rec {
+  pname = "libdnf";
+  version = "0.55.2";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = pname;
+    rev = version;
+    sha256 = "0hiydwfa90nsrqk5ffa6ks1g73wnsgjgq7z7gwq9jj76a7gpfbfq";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gettext
+    pkgconfig
+    check
+  ];
+
+  propagatedBuildInputs = [
```

Comment:
Why are all of these propagated?

Reply:
I might just misunderstand "propagatedBuildInputs".
But if I just use them as buildInputs, microdnf will complain:

```
-- Checking for module 'libdnf>=0.55.2'
--   Package 'librepo', required by 'libdnf', not found
```

Reply:
If the librepo really fails to build without this then it indeed should be propagated. But note that CMake tends to complain about stuff dependencies that are not actually needed.

Only dependencies listed in `Requires` in `result/lib/pkgconfig/libdnf.pc` (`glib-2.0`, `librepo` `rpm`, `sqlite3`) need to be propagated, those in `Requires.private` (`libsolv`, `libsolvext`) generally do not.

Reply:
ok, moved them aside from librepo, rpm, sqlite and libsolv

Using those, microdnf can compile now